signature/src/signatures/sigstore.rs

```diff
@@ -0,0 +1,254 @@
+// Copyright (c) 2021 Alibaba Cloud
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+use crate::image;
+use anyhow::{anyhow, Result};
+use oci_distribution::Reference;
+use serde::{Deserialize, Serialize};
+use std::collections::HashMap;
+use std::ffi::OsStr;
+use std::fs;
+
+// FIXME: Now, for Demo, the dir which the sigstore config files locate
+// is hardcoded to "/etc/containers/registries.d".
+const REGISTRY_CONFIG_DIR: &str = "/etc/containers/registries.d";
+
+// Format the sigstore name:
+// `image-repository@digest-algo=digest-value`
+#[allow(unused_assignments)]
+pub fn format_sigstore_name(image_ref: &Reference, image_digest: String) -> String {
```

Comment:
What if these parameters are empty/invalid? I think this function should check the values and return a `Result<String>` so that caller can be assured they are valid.